tests/unittests/unittest.cpp

```diff
@@ -1234,6 +1236,64 @@ void runCheckConsistencyTest(const std::string& filename) {
   REQUIRE_THAT(superfluous, UnorderedEquals<std::string>({"non-existant"}));
 }
 
+template <typename ReaderT, typename WriterT>
+void runRelationAfterCloneCheck(const std::string& filename = "unittest_relations_after_cloning.root") {
+  auto [hitColl, clusterColl, vecMemColl, userDataColl] = createCollections();
+  auto frame = podio::Frame();
+  frame.put(std::move(hitColl), "hits");
+  frame.put(std::move(clusterColl), "clusters");
+  frame.put(std::move(vecMemColl), "vectors");
+  // Empty relations
+  auto emptyColl = ExampleClusterCollection();
+  emptyColl.create();
+  emptyColl.create();
+  frame.put(std::move(emptyColl), "emptyClusters");
+  auto writer = WriterT(filename);
+  writer.writeFrame(frame, podio::Category::Event);
+  writer.finish();
+  auto reader = ReaderT();
+  reader.openFile(filename);
+  auto readFrame = podio::Frame(reader.readNextEntry(podio::Category::Event));
+
+  auto& clusters = readFrame.get<ExampleClusterCollection>("clusters");
+
+  auto nCluster = clusters[0].clone();
+  REQUIRE(nCluster.Hits().size() == 1);
+
+  auto hit = ExampleHit(420, {}, {}, {}, {});
+  nCluster.addHits(hit);
+  REQUIRE(nCluster.Hits().size() == 2);
+  REQUIRE(nCluster.Hits()[1].cellID() == 420);
+
+  auto nCluster2 = nCluster.clone();
+  REQUIRE(nCluster2.Hits().size() == 2);
+  auto anotherHit = ExampleHit(421, {}, {}, {}, {});
+  nCluster2.addHits(anotherHit);
+  REQUIRE(nCluster2.Hits().size() == 3);
+  REQUIRE(nCluster2.Hits()[2].cellID() == 421);
+
+  auto& vectors = readFrame.get<ExampleWithVectorMemberCollection>("vectors");
+  auto nVec = vectors[0].clone();
+  REQUIRE(nVec.count().size() == 2);
+  nVec.addcount(420);
+  REQUIRE(nVec.count().size() == 3);
+  REQUIRE(nVec.count()[2] == 420);
+
+  auto& emptyClusters = readFrame.get<ExampleClusterCollection>("emptyClusters");
+  auto nEmptyCluster = emptyClusters[0].clone();
+  REQUIRE(nEmptyCluster.Hits().empty());
+  nEmptyCluster.addHits(hit);
+  REQUIRE(nEmptyCluster.Hits().size() == 1);
+  REQUIRE(nEmptyCluster.Hits()[0].cellID() == 420);
+  nEmptyCluster.addHits(anotherHit);
+  REQUIRE(nEmptyCluster.Hits().size() == 2);
+  REQUIRE(nEmptyCluster.Hits()[1].cellID() == 421);
+}
+
+TEST_CASE("Relations after cloning with TTrees", "[ASAN-FAIL][UBSAN-FAIL][relations]") {
```

Comment:
```suggestion
TEST_CASE("Relations after cloning with TTrees", "[ASAN-FAIL][UBSAN-FAIL][relations][basics]") {
```
I would also add the `basics` tag here, since this is core functionality. Potentially even a `[memory_management]`.

Similar for the rntuple and sio based ones.

Reply:
Added